### ISSUE TITLE
Raspberry Pi4 Ubuntu no image found in manifest list for architecture arm64, variant v8, OS linux

### ISSUE THREAD

**Issue by stangerm2** [containers/image#1100]
Seems unresolved though tickets (containers/image/issues/898 cri-o/cri-o/issues/3626 containers/image/pull/904 cri-o/cri-o/issues/3233) are closed:

**Description**

After building crio on release-1.20 branch on Ubuntu 20 Raspberry Pi for aarch64/v8 kernel received:

```
sudo kubeadm init
[init] Using Kubernetes version: v1.20.0
[preflight] Running pre-flight checks
...
	[ERROR ImagePull]: failed to pull image k8s.gcr.io/kube-apiserver:v1.20.0: output: time="2020-12-15T16:47:28Z" level=fatal msg="pulling image: rpc error: code = Unknown desc = Error choosing image instance: no image found in manifest list for architecture arm64, variant v8, OS linux"
, error: exit status 1

ubuntu@ubuntu:~$ sudo crictl pull k8s.gcr.io/pause:3.1
FATA[0000] pulling image: rpc error: code = Unknown desc = Error choosing image instance: no image found in manifest list for architecture arm64, variant v8, OS linux 

ubuntu@ubuntu:~$ sudo crictl pull alpine:latest
Image is up to date for docker.io/library/alpine@sha256:a126728cb7db157f0deb377bcba3c5e473e612d7bafc27f6bb4e5e083f9f08c2
```

**Steps to reproduce the issue:**

Install Ubuntu 20 x64 bit kernel on Raspberry Pi 4
Build & install crio release-1.20 branch
And optionally install kubelet 1.20 with snap/src
crictl pull k8s.gcr.io/pause:3.1
crictl pull k8s.gcr.io/pause:3.2
kubeadm init

**Describe the results you received:**

```
# crictl pull k8s.gcr.io/pause:3.1
FATA[0000] pulling image failed: rpc error: code = Unknown desc = Error choosing image instance: no image found in manifest list for architecture arm, variant v8, OS linux
```
**Describe the results you expect:**
```
# crictl pull k8s.gcr.io/pause:3.1
Image is up to date for k8s.gcr.io/pause@sha256:c84b0a3a07b628bc4d62e5047d0f8dff80f7c00979e1e28a821a033ecda8fe53
```

ubuntu@ubuntu:~$ crio --version
INFO[0000] Starting CRI-O, version: 1.20.0, git: d388528dbed26b93c5bc1c89623607a1e597aa57(clean) 
crio version 1.20.0
Version:       1.20.0
GitCommit:     d388528dbed26b93c5bc1c89623607a1e597aa57
GitTreeState:  clean
BuildDate:     2020-12-14T23:00:59Z
GoVersion:     go1.14.7
Compiler:      gc
Platform:      linux/arm64
Linkmode:      dynamic

**Comment by mtrmac** [comment]
Thanks for your report.

This doesn’t make sense to me: AFAICS the code always tries a no-variant image as well, and testing locally
```
skopeo --override-os linux --override-arch arm64 --override-variant v8 inspect --config docker://k8s.gcr.io/pause:3.1
```
does find an image.

**Comment by stangerm2** [comment]
As a normal user, distro-dev I it's not readily apparent to me how I resolve or work around the issue. I'm reporting this because this is what you'd run into for the normal flow of an RiP4 Kub setup.

**Comment by mtrmac** [comment]
As a library author, “buy a computer” is not an easily accessible step to reproduce, and the above matches my understanding of the external environment.

Basically this probably requires a debugger or adding debug logging to the return value of `c/image/internal/pkg/platform.WantedPlatform` and a caller, probably `c/image/manifest/Schema2List.ChooseInstance` , to see how choosing the image fails.

---

Actually I suspect that the reported version number is incorrect: As of d388528dbed26b93c5bc1c89623607a1e597aa57 :
```console
% git grep 'no image found in manifest'  d388528dbed26b93c5bc1c89623607a1e597aa57  
d388528dbed26b93c5bc1c89623607a1e597aa57:vendor/github.com/containers/image/v5/manifest/docker_schema2_list.go: return "", fmt.Errorf("no image found in manifest list for architecture %s, variant %q, OS %s", wantedPlatforms[0].Architecture, wantedPlatforms[0].Variant, wantedPlatforms[0].OS)
```
Note the `%q`: this should fail with quotes around `"v8"`. The lack of quotes suggests that the failing implementation is pre-f0b7c36a56fc72b3927e4f30653ba1a08036728c , i.e. clearly pre-#904.

**Comment by stangerm2** [comment]
I built off:
`ubuntu@ubuntu:~/cri-o$ git log
commit d388528dbed26b93c5bc1c89623607a1e597aa57 (HEAD -> release-1.20, tag: v1.20.0, origin/release-1.20)
`

**Comment by stangerm2** [comment]
@mtrmac I assumed incorrectly that cri-o 1.20 had update to after #904 since it's been a while but they haven't. I rebuilt the lib and then cri-o and now the pull works. Please close sorry for the trouble, I'll go bug them now.

**Comment by mtrmac** [comment]
FWIW I do see #904 included in the quoted commit d388528dbed26b93c5bc1c89623607a1e597aa57 == `v1.20.0`. This looks as if some other version is actually running and serving the requests.

Or is that because Ubuntu/Debian has unbundled CRI-O and replaced the vendored version?

**Comment by stangerm2** [comment]
I'm certain I removed the Ubuntu 20 PPA that I installed, and the `crio --version` above show's the right commit for the bin I was running.
The debian pkg is only supporting up to 17.4 ATM https://download.opensuse.org/repositories/devel:/kubic:/libcontainers:/stable/ so you end up in this situation on arm64. Guess it's a pkg maintainer prob not yours.
